### ISSUE TITLE
Add option to generate chained assertions

### ISSUE THREAD

**Issue by MichaelBitard** [joel-costigliola/assertj-assertions-generator-maven-plugin#38]
Enhancement

Use case:

* You have an object "objectA" of type ObjectA which contains (among other fields) a field objectB of type ObjectB
* To assert it, I do
```java
assertThat(objectA).hasFieldOfObjectA("field");
assertThat(objectA.objectB).hasFieldOfObjectB("fieldOfObjectB");
```
* I'd like to be able to assert like this:
```java
assertThat(objectA)
  .hasFieldOfObjectA("field")
  .withObjectB()
    .hasFieldOfObjectB("fieldOfObjectB");
```
I already do that manually, but I think it could be useful to be able to add this behavior automatically.

What do you think?

**Comment by joel-costigliola** [comment]
I agree on the idea, I would just use `getB()` to navigate to the B's assertions.

**Comment by yoorick** [comment]
I would rather prefer to see it as `has<FieldName>Satisfying(Consumer/Condition)` pair of methods. Just like [hasValueSatisfying](https://github.com/joel-costigliola/assertj-core/blob/31ffc245f8371b53b65ed93820a5a4f91067e6d7/src/main/java/org/assertj/core/api/AbstractOptionalAssert.java#L167) for Optional or [satisfies](https://github.com/joel-costigliola/assertj-core/blob/31ffc245f8371b53b65ed93820a5a4f91067e6d7/src/main/java/org/assertj/core/api/AbstractListAssert.java#L188) for list elements or other similar asserts. This way you can continue to fluently verify other complex fields. With `getB()/withObjectB()` approach you have no way to fluently return back to objectA assertions.

**Comment by yoorick** [comment]
Something like this:
```java
assertThat(objectA)
    .hasFieldOfObjectA("field")
    .hasObjectBSatisfying(
        objectB -> assertThat(objectB)
            .hasFieldB1("value1")
            .hasFieldB2("value2")
    )
    .hasObjectCSatisfying(
        objectC -> assertThat(objectC)
            .hasFieldC1("otherValue1")
            .hasFieldC2(1234)
            .hasFieldC3("otherValue3")
    )
    // ...
